### PR TITLE
HR Data Science Project (first half)

### PR DESCRIPTION
**Pull request by Eugene3819**

### CODE REVIEW

**Comment by alexnaylor1999** [comment]
Hi @Eugene3819, this is a really good start. Have you had a chance to work on the second half of the project? :)

File: EugeneChai/hr analytics.Rmd [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+```
+
+
+# Data Imputation
+```{r}
+colSums(is.na(df))
+```
+```{r}
+#The null values in NumCompaniesWorked can be filled by using estimated values formed by using both YearsAtCompany and TotalWorkingYears.
+na_numcompaniesworked<-df %>% filter(is.na(NumCompaniesWorked))%>%select(EmployeeID,NumCompaniesWorked,YearsAtCompany,TotalWorkingYears)
```
Comment (Stephen-Cole267) [inline EugeneChai/hr analytics.Rmd:120]: Does this use YearsAtCompany and TotalWorkingYears to impute the nulls? This seems like a very interesting approach, I'm curious as to your thinking behind this :)

File: EugeneChai/hr analytics.Rmd [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+```{r}
+#To replace null values for TotalWorkingYears,estimated values are calculated using NumCompaniesWorked and TotalWorkingYears
+df %>% filter(is.na(TotalWorkingYears))%>%select(EmployeeID,NumCompaniesWorked,YearsAtCompany,TotalWorkingYears)
+```
+```{r}
+#Manually fill in values where NumCompaniesWorked is equal to 0 or 1
+df[2368,20]<-10
+df[4410,20]<-21
+df[24,20]<-25
+df[3819,20]<-6
```
Comment (Stephen-Cole267) [inline EugeneChai/hr analytics.Rmd:159]: Could this be done in a function? This is me being very nitpicky btw and I like to automate anything where possible 👍🏼

**Comment by Stephen-Cole267** [comment]
In Summary, Im very happy with the approach. Good analysis on each feature, created your own features, and easy to follow code; a great first half in all. Looking forward to seeing the second half. 

I can't give much advice around how to improve the code itself, as I'm really rusty with R, but some exploration around the attrition target feature would be useful. Could look into how some of the explanatory variables could possibly contribute to a positive or negative case by looking at correlation and plotting some charts.

Could also look at fitting different types of models. This would allow you to ensure that the correct model has been chosen and would also be good practice for model comparisons when looking at the different metrics. If you already do that in the second half then ignore me!

**Comment by Eugenechai38** [comment]
> Hi @Eugene3819, this is a really good start. Have you had a chance to work on the second half of the project? :)

Hi Alex, sorry for the late reply. I'm currently redoing the project in Python.

**Pull request closed without merging**